velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -41,6 +30,78 @@ void ensureRegexIsConstant(
   }
 }
 
+FOLLY_ALWAYS_INLINE std::string prepareRegexpPattern(
```

Comment:
Are we processing the pattern from Spark to ensure it is compatible with RE2 so as to get a consistent result with Spark? Perhaps add some comments for these helper functions to illustrate the purpose and how we achieve that.